tools/setup/emoji/emoji_setup_utils.py

```diff
@@ -0,0 +1,179 @@
+    'mega', # megaphone
+    'ski', # skis
+    'high_heel', # high_heeled_shoe (so that it shows up when searching for shoe)
+    # less confident about the following
+    'dolls', # japanese_dolls
+    'moon', # waxing_gibbous_moon (should rename crescent_moon to moon)
+    'clapper', # clapper_board
+    'traffic_light', # horizontal_traffic_light (there's also a vertical_traffic_light)
+    'lantern',
+    'red_paper_lantern', # izakaya_lantern (in the future we should make sure
+                         # red_paper_lantern finds this)
+
+    # would be chosen by longer
+    'down_button', # arrow_down_small, I think to match the other arrow_*
+                   # names. Matching what github and slack do.
+    'running_shoe', # athletic_shoe, both github and slack agree here.
+    'running', # runner. slack has both, github has running_man and running_woman, but not runner
+    'o2', # o_button
+    'star2', # glowing_star
+    'bright', # high_brightness, to match low_brightness, what github/slack do
+    'dim_button', # low_brightness, copying github/slack
+    'stars', # shooting_star. disagrees with github, slack, and emojione, but this seems better
+    'nail_care', # nail_polish. Also disagrees github/slack/emojione, is nail_polish mostly an
+                 # american thing?
+    'busstop', # bus_stop
+    'tophat', # top_hat
+    'old_woman', # older_woman, following github/slack/emojione on these
+    'old_man', # older_man
+    'blue_car', # recreational_vehicle
+    'litter_in_bin_sign', # put_litter_in_its_place
+    'moai', # moyai based on github/slack
+    'fuelpump', # fuel_pump
+
+    # names not otherwise excluded by our heuristics
+    'left_arrow', # arrow_left, to match other arrow_* shortnames
+    'right_arrow', # arrow_right
+    'up_arrow', # arrow_up
+    'down_arrow', # arrow_down
+    'chequered_flag', # checkered_flag
+    'e_mail', # e-mail
+    'non_potable_water', # non-potable_water
+    'flipper', # dolphin
+]
+
+## functions that take in a list of names at a codepoint and return a subset to exclude
+
+def blacklisted(names):
+    # type: (List[str]) -> List[str]
+    return [name for name in names if name in blacklisted_names]
+
+# 1 letter names don't currently show up in our autocomplete. Maybe should
+# change our autocomplete so that a whitelist of letters do, like j (for joy), x, etc
+# github uses a, ab, etc. instead of a_button, slack doesn't have any of the [letter]_buttons
+def one_lettered(names):
+    # type: (List[str]) -> List[str]
+    if len(names) == 1:
+        return []
+    return [name for name in names if len(name) == 1]
+
+# If it is an ideograph (or katakana, but we'll probably deal with that
+# differently after 1.5), remove any names that don't have
+# ideograph/katakana in them
+def ideographless(names):
+    # type: (List[str]) -> List[str]
+    has_ideographs = ['ideograph' in name.split('_') or
+                      'katakana' in name.split('_') for name in names]
```

Comment:
You probably want to use the `any()` function here (maps a list to whether anything appears in the list).